File: April2022/Java/ContainerWithMostWater.java

// @saorav21994
// TC : O(n)
// SC : O(1)

class Solution {
    public int maxArea(int[] height) {
        int l = height.length-1;
        int max = Integer.MIN_VALUE;
        int i = 0;
        while (i < l) {
            max = Math.max(max, Math.min(height[i], height[l]) * (l-i));
            if (height[i] < height[l])
                i += 1;
            else 
                l -= 1;
        }
        return max;
    }
}

File: April2022/C++/Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0,r=height.size()-1;
        int res=0;
        while(l<r)
        {
            res=max(res,(r-l )* min(height[l],height[r]));
            if(height[l]<height[r]) l++;
            else r--;
        }
        return res;
    }
};